lib/ansible/modules/windows/win_disk_management.ps1

```diff
@@ -0,0 +1,1411 @@
+        )
+
+        $null = $Disk| Initialize-Disk -PartitionStyle $PartitionStyle -Confirm:$false
+
+        return
+}
+
+function Convert-PartitionStyle {
+        param(
+                $Disk,
+                $PartitionStyle
+        )
+
+        $null = Invoke-Expression "'Select Disk $($Disk.Number)','Convert $($PartitionStyle)' | diskpart"
+
+        return
+}
+
+function Manage-ShellHWService {
+        param(
+                $action
+        )
+
+        switch ($action) {
+                Stop {
+                    $null = Stop-Service -Name ShellHWDetection
+
+                    return
+                }
+                Start {
+                    $null = Start-Service -Name ShellHWDetection
+
+                    return
+                }
+                Check {
+                    $CheckService = (Get-Service ShellHWDetection).Status -eq "Running"
+
+                    return $CheckService
+                }
+                default {
+                    throw "Neither '-Stop', '-Start' nor 'Check' switch was passed to the function when invoked, without the switches the service can not be maintained"
+                }
+        }
+}
+
+function Create-Partition {
+        param(
+                $Disk,
+                $SetDriveLetter
+        )
+
+        $Partition = $Disk | New-Partition -UseMaximumSize -DriveLetter $SetDriveLetter
+
+        return $Partition
+}
+
+function Create-Volume {
+        param(
+                $Volume,
+                $FileSystem,
+                $FileSystemLabel,
+                $FileSystemAllocUnitSize,
+                $FileSystemLargeFRS,
+                $FileSystemShortNames,
+                $FileSystemIntegrityStreams
+        )
+
+        $Alloc = $FileSystemAllocUnitSize *1KB
+
+        $ParaVol = @{
+            FileSystem = $FileSystem
+            NewFileSystemLabel = $FileSystemLabel
+            AllocationUnitSize = $Alloc
+        }
+
+        if ($FileSystem -eq "ntfs") {
+            if (-not $FileSystemLargeFRS) {
+                $CreatedVolume = $Volume | Format-Volume @ParaVol -ShortFileNameSupport $FileSystemShortNames -Force -Confirm:$false
+
+                return $CreatedVolume
+            } else {
+                $CreatedVolume = $Volume | Format-Volume @ParaVol -UseLargeFRS -ShortFileNameSupport $FileSystemShortNames -Force -Confirm:$false
+
+                return $CreatedVolume
+            }
+        } elseif ($FileSystem -eq "refs") {
+            $CreatedVolume = $Volume | Format-Volume @ParaVol -SetIntegrityStreams $FileSystemIntegrityStreams -Force -Confirm:$false
+
+            return $CreatedVolume
+        }
+}
+
+# Rescan disks
+try {
+    $null = Invoke-Expression '"rescan" | diskpart'
+} catch {
+    $result.general_log.rescan_disks = "failed"
+}
+$result.general_log.rescan_disks = "successful"
+
+$ParamsDisk = @{
+    DiskSizeInteger = $SizeInteger
+    DiskNumberInteger = $NumberInteger
+    DiskSize = "$(if ($SizeInteger) {
```

Comment:
Move this outside of this definition to make it easier to understand what is going wrong here

Reply:
replaced it